### PR TITLE
feat(user): create authentification system

### PR DESCRIPTION
**Pull request by justinDev91**
# Ticket Notion

# Screen (si besoin)

# Autre information

### CODE REVIEW

File: src/assets/controllers.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "controllers": [],
+    "entrypoints": []
+}
```
Comment (MisterAzix) [inline src/assets/controllers.json:4]: Not relative to the PR ?

File: src/assets/controllers/hello_controller.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * this controller to be executed. The name "hello" comes from the filename:
+ * hello_controller.js -> "hello"
+ *
+ * Delete this file or adapt it for your use!
+ */
+export default class extends Controller {
+    connect() {
+        this.element.textContent = 'Hello Stimulus! Edit me in assets/controllers/hello_controller.js';
+    }
+}
```
Comment (MisterAzix) [inline src/assets/controllers/hello_controller.js:16]: Not relative to the PR ?

File: src/assets/countVote.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    let answer_id = $(e.currentTarget).data('id')
+    let pill = $(e.currentTarget).parent().find('.js-vote-total')
+
+    $.post(
+        '/answers/'+answer_id + '/vote', {direction : direction}, function (response) {
+            pill.text(response.votes)
+        }
+    )
+})
+
```
Comment (MisterAzix) [inline src/assets/countVote.js:15]: Not relative to the PR ?

File: src/src/DataFixtures/AppFixtures.php [hunk 4]
```diff
@@ -2,16 +2,44 @@
...
+        UserFactory::createMany(10);
+        TagFactory::createMany(100);
+        //Créer 10 questions, excuter symfony console doctrine:fixtures:load pour charger en db
+        QuestionFactory::createMany(10);
+        //Je passe les questions à une réponse
+        AnswerFactory::createMany(50);
+        //Les questions non publiées
+        //QuestionFactory::new()->notPublished()->many(5)->create();
+        AnswerFactory::new()->needsApproval()->many(20)->create();
+        AnswerFactory::new()->spam()->many(10)->create();
```
Comment (MisterAzix) [inline src/src/DataFixtures/AppFixtures.php:43]: Not relative to the PR 👀 
```suggestion
```

File: src/src/DataFixtures/AppFixtures.php [hunk 5]
```diff
@@ -2,16 +2,44 @@
 
 namespace App\DataFixtures;
 
+use App\Entity\Question;
+use App\Factory\AnswerFactory;
+use App\Factory\QuestionFactory;
+use App\Factory\TagFactory;
```
Comment (MisterAzix) [inline src/src/DataFixtures/AppFixtures.php:8]: Not relative to the PR
```suggestion
```

File: src/src/Controller/UserController.php [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        ]);
+    }
+
+    #[Route('/user/new', name: 'app_user_new')]
+    public function addUser(): Response
+    {
+        return $this->render('user/user_new.html.twig');
+    }
+
+    #[Route('/user/create', name: 'app_user_create_api', methods: 'POST')]
```
Comment (MisterAzix) [inline src/src/Controller/UserController.php:33]: Why `/user/create` and not just `/user/new` why POST method ?

File: src/src/Controller/UserController.php [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+    #[Route('/user/{email}/modify', name: 'app_user_modify')]
+    public function modify(User $user): Response
+    {
+        return $this->render('user/user_modify.html.twig', [
+            'user' => $user
+        ]);
+    }
+
+    #[Route('/user/{email}/update', name: 'app_user_update_api', methods: 'POST')]
```
Comment (MisterAzix) [inline src/src/Controller/UserController.php:75]: Same why `/user/{email}/update` and not `/user/{email}/modify` with POST method ?

File: src/src/Entity/User.php [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+    private ?string $email = null;
+
+    #[ORM\Column(type: Types::DATETIME_MUTABLE, nullable: true)]
+    private ?\DateTimeInterface $createDate = null;
+
+    #[ORM\OneToMany(mappedBy: 'user', targetEntity: question::class, orphanRemoval: true)]
+    private Collection $question;
+
+    #[ORM\OneToMany(mappedBy: 'user', targetEntity: answer::class)]
+    private Collection $answer;
```
Comment (MisterAzix) [inline src/src/Entity/User.php:39]: Delete if possible to delete Answer and Question entities to have a clean PR ;)

File: src/src/Form/QuestionFormType.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (MisterAzix) [inline src/src/Form/QuestionFormType.php:1]: Reminder for me @MisterAzix

File: src/templates/api/index.html.twig [hunk 10]
```diff
@@ -0,0 +1,20 @@
+{% extends 'base.html.twig' %}
```
Comment (MisterAzix) [inline src/templates/api/index.html.twig:1]: Is this file usefull ?

File: src/src/Controller/ApiController.php [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+class ApiController extends AbstractController
+{
+    #[Route('/api', name: 'app_api')]
+    public function index(): Response
+    {
+        return $this->json([
+            'message' => "Hello ord"
+        ]);
+    }
+}
```
Comment (MisterAzix) [inline src/src/Controller/ApiController.php:18]: Is this file usefull ?

File: src/migrations/Version20221023115039.php [hunk 12]
```diff
@@ -0,0 +1,51 @@
+<?php
```
Comment (MisterAzix) [inline src/migrations/Version20221023115039.php:1]: Is it possible to delete all migrations files and recreate them to only have what's relative with User ? ;)

**Review by MisterAzix — COMMENTED** [review]
Normally ur PR is only relative to User, so u need to only have files relative with this feature, on the other hand good job it seems to be really good 🥳

**Comment by MisterAzix** [comment]
> Normally ur PR is only relative to User, so u need to only have files relative with this feature, on the other hand good job it seems to be really good 🥳

Wanting to get ahead is not a problem at all, on the contrary. You just have to leave some tasks to others :eyes:

File: src/assets/countVote.js (hunk 3 above)
Comment (justinDev91) [reply src/assets/countVote.js:15]: Code generate by webpack to use nodeJS

File: src/src/DataFixtures/AppFixtures.php (hunk 4 above)
Comment (justinDev91) [reply src/src/DataFixtures/AppFixtures.php:43]: Yeah but we need this to generate fake data anywhere. Somebody has to do thats so...

**Comment by MisterAzix** [comment]
**Pro tips :** with `git rebase -i origin/dev` u can select what commit of your branch u want to keep 👀 

(ask me if u want some help with)

File: src/src/Entity/User.php (hunk 8 above)
Comment (justinDev91) [reply src/src/Entity/User.php:39]: ok done

File: src/templates/api/index.html.twig (hunk 10 above)
Comment (justinDev91) [reply src/templates/api/index.html.twig:1]: Not too much ! let me delete it

File: src/src/Form/QuestionFormType.php (hunk 9 above)
Comment (justinDev91) [reply src/src/Form/QuestionFormType.php:1]: Oui chef @MisterAzix

File: src/migrations/Version20221023115039.php (hunk 12 above)
Comment (justinDev91) [reply src/migrations/Version20221023115039.php:1]: Yeah all migration is automatically  deleted when u push in githuh everyone should type symfony console make:migration then doctrine:migration:migrate @MisterAzix

File: src/src/DataFixtures/AppFixtures.php (hunk 5 above)
Comment (justinDev91) [reply src/src/DataFixtures/AppFixtures.php:8]: @MisterAzix

File: src/src/DataFixtures/AppFixtures.php (hunk 4 above)
Comment (justinDev91) [reply src/src/DataFixtures/AppFixtures.php:43]: @MisterAzix

File: src/assets/countVote.js (hunk 3 above)
Comment (justinDev91) [reply src/assets/countVote.js:15]: @MisterAzix

File: src/assets/controllers/hello_controller.js (hunk 2 above)
Comment (justinDev91) [reply src/assets/controllers/hello_controller.js:16]: code generate by webpack to use nodeJS et js @MisterAzix

**Comment by justinDev91** [comment]
Normally everything is fine

File: src/src/Controller/UserController.php (hunk 6 above)
Comment (justinDev91) [reply src/src/Controller/UserController.php:33]: Post method is required haha, yeah create or new doesn’t change smtg. theres a new fand create is not the same

File: src/assets/controllers.json (hunk 1 above)
Comment (justinDev91) [reply src/assets/controllers.json:4]: Generally generated by webpack

Comment (justinDev91) [reply src/assets/controllers.json:4]: @MisterAzix

File: src/migrations/Version20221023115039.php (hunk 12 above)
Comment (MisterAzix) [reply src/migrations/Version20221023115039.php:1]: Mmmmh sure ? so why is there 2 migrations file here ? 😅

File: src/src/DataFixtures/AppFixtures.php (hunk 4 above)
Comment (MisterAzix) [reply src/src/DataFixtures/AppFixtures.php:43]: Sure, but gonna be add by the person who create the feature

File: src/migrations/Version20221023115039.php (hunk 12 above)
Comment (justinDev91) [reply src/migrations/Version20221023115039.php:1]: Haha tu gères

File: src/src/DataFixtures/AppFixtures.php (hunk 5 above)
Comment (justinDev91) [reply src/src/DataFixtures/AppFixtures.php:8]: Pas grave . t'façon j'aurai besoin de ca

File: src/src/DataFixtures/AppFixtures.php (hunk 4 above)
Comment (justinDev91) [reply src/src/DataFixtures/AppFixtures.php:43]: Yeah sure ! je vais delete d'autres

**Pull request merged**